File: Phase1_Challenge/NodeSwap.java
class Node{
    int data;
    Node next;
    
    Node(int data){
        this.data = data;
        this.next = null;
    }
}

public class NodeSwap{
    static Node solve(Node head, int X){
    if(head == null || head.next == null){
            return head;
        }
        
        if(head.data == X){
            Node temp = head.next;
            head.next = temp.next;
            temp.next = head;       
            head = temp;
        }

        Node prev = head;
        Node curr = head.next;
        
        while(curr != null && curr.next != null){  
            if(curr.data == X){
                Node temp = curr.next;
                curr.next = temp.next;
                prev.next = temp;
                temp.next = curr;
                prev = temp;
            } else {
                prev = curr;
                curr = curr.next;
            }
        }
        return head;
    }
    public static void main(String[] args) {
        Node head = new Node(5);
        head.next = new Node(6);


        int X = 6;
        head = solve(head, X);

        Node curr = head;
        while(curr != null){
            System.out.print(curr.data + " ");
            curr = curr.next;
        }
    }
}

File: Challenge_Phase1/nodeSwap.cpp
#include<bits/stdc++.h>
using namespace std;

class Node{
    public:
    int val;
    Node* next;
    Node(int val){
        this->val=val;
        next=NULL;
    }
};
void swapNode(Node *p,Node *q,Node *prev){
    Node *temp=new Node(0);
    p->next=q->next;
    prev->next=q;
    q->next=p;
}
Node *swapbyValue(Node* head,int x){
    if(head==NULL || head->next==NULL) return head;
    if(head->val==x){
        Node* first=head;
        Node* second=head->next;
        first->next=second->next;
        second->next=first;
        head=second;
    }
    Node* prev=NULL;
    Node* curr=head;
    Node* nxt=head->next;
    while(nxt!=NULL){
        if(curr->val==x){
            swapNode(curr,nxt,prev);
            curr=curr->next;
            nxt=curr->next;
        }
        else{
            prev=curr;
            curr=nxt;
            nxt=nxt->next;
        }
    }
    return head;
}

int main(){
    int n;
    cout<<"Enter number of nodes:";
    cin>>n;
    Node* head=NULL;
    Node* tail=NULL;
    cout<<"Enter node values:"<<endl;
    for(int i=0;i<n;i++){
        int val;
        cin>>val;
        Node* newNode=new Node(val);
        if(head==NULL){
            head=newNode;
            tail=newNode;
        }
        else{
            tail->next=newNode;
            tail=newNode;
        }
    }

    cout<<"Original Linked List:"<<endl;
    Node* temp=head;
    while(temp!=NULL){
        cout<<temp->val<<" ";
        temp=temp->next;
    }
    cout<<endl;
    cout<<"Enter value to swap:";
    int x;
    cin>>x;
    head=swapbyValue(head,x);
    cout<<"Linked List after swapping:"<<endl;
    temp=head;
    while(temp!=NULL){
        cout<<temp->val<<" ";
        temp=temp->next;
    }
    return 0;
}